ls1intum/Artemis
# `Programming exercise`: Fix inactive test cases being assigned to wrong task

### Checklist
#### General


#### Server
#### Changes affecting Programming Exercises

### Motivation and Context


There are some use cases where instructors use inactive test cases. 
To be consistent with the new additional of task based grading we fix this edge case to also work based on tasks.

### Description

Fixes an edge case where inactive test cases are no longer associated with their task

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Create an exercise with the default settings.
1. Go to the exercise grading view, take a look at the view and remember the visible tasks/test cases or make a screenshot
1. Go to the exercise management and click on `Edit in editor`
1. Select the test repository and delete some test cases, making them inactive (e.g. behavioral)
1. Go back to the grading view and see if the test cases are still in their tasks

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage



| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| ProgrammingExerciseTaskService | 88.77% | ✅              |

### Screenshots
